src/main/java/com/nvs/store/service/CartService.java

```diff
@@ -0,0 +1,202 @@
+package com.nvs.store.service;
+
+import com.nvs.store.dto.*;
+import com.nvs.store.models.cart.Cart;
+import com.nvs.store.models.cart.CartItem;
+import com.nvs.store.models.product.Product;
+import com.nvs.store.models.user.User;
+import com.nvs.store.repository.*;
+import com.nvs.store.util.CartUtils;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class CartService {
+    private final CartRepository cartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final UserRepository userRepository;
+    private final ProductRepository productRepository;
+
+    public CartDTO findActualCartByEmail() {
+        String userEmail = getUserEmailFromContext();
+        User user = userRepository
```

Comment:
You can simply the get of actual cart by writing a join sql between cart and user. By simplifying the get you will improve the code and the performance where isn't required to extract unuseful data of user from database and will avoid additional call to database. 
You can write following sql: `select c.* from cart c join user u on c.user_id = c.id where u.email = :email and c.actual = true.` Hibernate will map the result of the sql query to Cart entity object.